x/ccv/provider/keeper/proposal.go

```diff
@@ -29,6 +30,75 @@ func (k Keeper) CreateConsumerChainProposal(ctx sdk.Context, p *types.CreateCons
 	return nil
 }
 
+// StopConsumerChainProposal stops a consumer chain and released the outstanding unbonding operations.
+// If the stop time hasn't already passed, it stores the proposal as a pending proposal.
+func (k Keeper) StopConsumerChainProposal(ctx sdk.Context, p *types.StopConsumerChainProposal) error {
+	if ctx.BlockTime().After(p.StopTime) {
```

Comment:
The `After` could make it a PITA in the ass to test. Would having `!Before` cause a mismatch with the spec?

(cc @mpoke )